core/rio/ntriples/src/main/java/org/eclipse/rdf4j/rio/ntriples/NTriplesParser.java

```diff
@@ -211,9 +211,16 @@ protected int skipToEndOfLine(int c) throws IOException {
 	 * Reads characters from reader until the first EOL has been read. The first character after the EOL is returned. In
 	 * case the end of the character stream has been reached, -1 is returned.
 	 */
-	protected int skipLine(int c) throws IOException {
+	protected int skipLine(int c, StringBuilder sb) throws IOException {
 		while (c != -1 && c != '\r' && c != '\n') {
 			c = readCodePoint();
+			if (sb != null) {
+				sb.append(Character.toChars(c));
```

Comment:
I just had a closer look and you get a test failure because this does not check if c (after reading the code point on line 216) is -1.